labonneboite/common/maps/README.md

```diff
@@ -0,0 +1,360 @@
+
+
+`labonneboite/web/search/views.py`
+
+```
+@searchBlueprint.route('/entreprises')
+def entreprises():
+    # ...
+    # Anticipate future calls by pre-computing duration-related searches
+    if location:
+        precompute.isochrones((location.latitude, location.longitude))
+```
+
+It means that our app retrieves isochrone data for **each two transport modes** and for **each four durations** specified in `labonneboite/common/maps/constants`.
+
+`labonneboite/common/maps/precompute.py`
+
+```
+def isochrones(location):
+    """
+    Compute isochrones asynchronously for all durations and modes. Each isochrone
+    is computed in a separate task.
+    """
+    for mode in travel.TRAVEL_MODES:
+        for duration in constants.ISOCHRONE_DURATIONS_MINUTES:
+            isochrone(location, duration, mode=mode)
+```
+
+:warning: Isochrone data is a list of polygons (a list of lists of coordinates), not Elastic Search results! More precisely, its form is:
+
+```
+[
+    [
+        ({latitude}, {longitude}),
+        ({latitude}, {longitude})
+    ],
+    [...]
+]
+
+# Example: `[[(3.504869, 45.910195), (3.504860, 45.910194)], [...]]`
+```
+
+:information_source: Data is stored in a cache (see [section Cache](#cache) for further information). The `isochrone` task checks if there is already data for this location in the cache. If not, it stores it.
+
+
+#### 2/ Use this data with Elastic Search
+
+Elastic Search understands pretty well geometric data, including polygons. We just have to pass it to the Elastic Search query. This is made in two steps:
+
+`labonneboite/web/search/views.py`
+
+```
+@searchBlueprint.route('/entreprises')
+def entreprises():
+    # Convert request arguments to fetcher parameters
+    parameters = get_parameters(request.args)
+
+    # Fetch offices and alternatives
+    # Other filters are hidden in this documentation for readability purposes.
+    fetcher = search_util.HiddenMarketFetcher(
+        location,
+        travel_mode=parameters['travel_mode'],
+        duration=parameters['duration'],
+        # ...
+    )
+```
+
+
+`labonneboite/common/search.py`
+
+```
+def build_json_body_elastic_search(...):
+    # ...
+    # Add a filter in Elastic Search
+    if duration is not None:
+        # Retrieve cached data
+        isochrone = travel.isochrone((latitude, longitude), duration, mode=travel_mode)
+        if isochrone:
+            for polygon in isochrone:
+                should_filters.append({
+                    "geo_polygon": {
+                        "locations": {
+                            "points": [[p[1], p[0]] for p in polygon]
+                        }
+                    }
+                })
+```
+
+
+#### 3/ Display filtered results to the user
+
+Once results are ready in the `search.entreprises` view, two templates can be shown:
+- if request is an AJAX call: `search/results_content.html`
+- if not: `search/results.html`
```

Comment:
This is not specific to isochrone requests. It is necessary to refresh results, e.g: when a user moves the map with the "refresh results when I move the map" checkbox enabled.

Reply:
Information added!